jetty-client/src/main/java/org/eclipse/jetty/client/RoundRobinConnectionPool.java

```diff
@@ -43,36 +55,16 @@ public RoundRobinConnectionPool(HttpDestination destination, int maxConnections,
     public RoundRobinConnectionPool(HttpDestination destination, int maxConnections, Callback requester, int maxMultiplex)
     {
         super(destination, maxConnections, false, requester, maxMultiplex);
-        pool = destination.getBean(Pool.class);
-    }
-
-    @Override
-    protected Connection acquire(boolean create)
-    {
         // If there are queued requests and connections get
         // closed due to idle timeout or overuse, we want to
         // aggressively try to open new connections to replace
         // those that were closed to process queued requests.
-        return super.acquire(true);
+        setMaximizeConnections(true);
     }
 
     @Override
-    protected Connection activate()
+    protected int getIndex(int maxConnections)
     {
-        Pool<Connection>.Entry entry;
-        try (Locker.Lock l = lock.lock())
-        {
-            int index = Math.abs(offset % pool.getMaxEntries());
-            entry = pool.acquireAt(index);
-            if (LOG.isDebugEnabled())
-                LOG.debug("activated at index={} entry={}", index, entry);
-            if (entry != null)
-                ++offset;
-        }
-        if (entry == null)
-            return null;
-        Connection connection = entry.getPooled();
-        acquired(connection);
-        return connection;
+        return Math.abs(offset.getAndIncrement() % maxConnections);
```

Comment:
When the integer goes overflows, you will start going the opposite way around the the pool

Reply:
could always use the overflow throwing behavior of `Math.addExact(int, int)` to trap that.

Reply:
Is that a problem?

Reply:
It might not be a problem, and unlikely to happen anyway.... but easily fixed with:
```
            return index.getAndUpdate(c -> Math.max(0, c + 1)) % maxConnections;
```

Reply:
And I have an even better solution in the latest code.